### DISCUSSION TITLE
(Feature request) How do I change the drive where the notes are stored?

### DISCUSSION THREAD

**Discussion by inikishev** [TriliumNext/Trilium#7366] [Q&A] [answered]
### Describe feature

Currently the notes seem to be stored on C drive and in some weird hidden folder as well (appdata/roaming). I would like my notes to be on drive D as it my portable drive. I tried moving the app itself to D, but weirdly that didn't work. For some reason it is still trying to access that strange hidden folder on C drive. Unfortunately I couldn't find anything useful by googling and by searching through app settings. I am using windows 11

**Comment by zadam** [comment, marked as answer]
Hi, the "weird hidden folder" is just following the official Microsoft guidelines on where the application data should be stored.

The directory can be changed via `TRILIUM_DATA_DIR` environment variable, there's some more info in wiki: https://github.com/zadam/trilium/wiki/Data-directory

**Comment by inikishev** [comment]
> Hi, the "weird hidden folder" is just following the official Microsoft guidelines on where the application data should be stored.
> 
> The directory can be changed via `TRILIUM_DATA_DIR` environment variable, there's some more info in wiki: https://github.com/zadam/trilium/wiki/Data-directory

Thank you, it worked, although I had to move all files from hidden directory into the new one. May I ask, I've never seen any apps saving actual documents into app data, why not save them into my documents or something? I am assuming app data is for configuration that users aren't supposed to access, hence it is hidden

**Comment by zadam** [comment]
> I've never seen any apps saving actual documents into app data, why not save them into my documents or something?

Even though the terminology is "document", it's not something fitting into the Documents folder - you won't be able to open it with a Word or similar apps.

>  I am assuming app data is for configuration that users aren't supposed to access, hence it is hidden

The App Data/Roaming is meant to contain application and user specific data which should "roam" with user. That's the case with trilium data.

**Comment by Nriver** [comment]
Actually, there are whole bunch of apps store data in those directories. For example, Evernote uses C:\Users\username\AppData\Local\Evernote.
